### PR TITLE
Reload Javascript

### PR DESCRIPTION
**Pull request by nmartinii**

### CODE REVIEW

File: library/Icinga/Authentication/Auth.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ public function setAuthenticated(User $user, $persist = true)
         } else {
             $preferences = new Preferences();
         }
-        // TODO(el): Quick-fix for #10957. Only reload CSS if the theme changed.
+        // TODO(el): Quick-fix for #10957. Only reload CSS if the theme changed. check
```
Comment (lippserd) [inline library/Icinga/Authentication/Auth.php]: I think this change slipped in. Please fix.

File: library/Icinga/Web/Controller/ActionController.php [hunk 2]
```diff
@@ -68,6 +68,8 @@ class ActionController extends Zend_Controller_Action
 
     protected $reloadCss = false;
 
+    protected $reloadJs = false;
```
Comment (lippserd) [inline library/Icinga/Web/Controller/ActionController.php]: Add PHPDoc.

File: library/Icinga/Web/Controller/ActionController.php [hunk 3]
```diff
@@ -240,6 +242,12 @@ protected function reloadCss()
         return $this;
     }
 
+    protected function reloadJs()
```
Comment (lippserd) [inline library/Icinga/Web/Controller/ActionController.php]: Add PHPDoc.

File: library/Icinga/Web/Controller/ActionController.php [hunk 4]
```diff
@@ -240,6 +242,12 @@ protected function reloadCss()
...
+    {
+        $this->reloadJs = true;
+        return $this;
```
Comment (lippserd) [inline library/Icinga/Web/Controller/ActionController.php]: Add newline before return.

File: library/Icinga/Web/Response.php [hunk 5]
```diff
@@ -198,6 +205,30 @@ public function getRequest()
...
+     * Set whether to instruct client side script code to reload JS
+     *
+     * @param   bool    $reloadJs
+     *
+     * @return  $this
+     */
+    public function setReloadJs($reloadJs)
+    {
+        $this->reloadJs = $reloadJs;
+        return $this;
```
Comment (lippserd) [inline library/Icinga/Web/Response.php]: Add newline before return.

File: library/Icinga/Web/Response.php [hunk 6]
```diff
@@ -198,6 +205,30 @@ public function getRequest()
...
+     */
+    public function isReloadJs()
+    {
+        return $this->reloadJs;
+    }
+
+    /**
+     * Set whether to instruct client side script code to reload JS
+     *
+     * @param   bool    $reloadJs
```
Comment (lippserd) [inline library/Icinga/Web/Response.php]: Please remove the extra spaces from PHPDoc. I know we in indented everything in the past, but it is enough to add alignment around parameter names (if there is more than one), i.e.

```php
/**
 * Example doc
 *
 * @param int    $short    Something short
 * @param string $fullname Full name
 *
 * @return $this
 */
function foobar($short, $fullname)
{
    return $this;
}
```

File: public/js/icinga.js [hunk 7]
```diff
@@ -226,6 +226,7 @@
                     window.icinga = new window.Icinga(oldConfig);
                     window.icinga.initialize();
                     window.icinga.ui.reloadCss();
+                    window.icinga.ui.reloadJs();
```
Comment (lippserd) [inline public/js/icinga.js]: Why?

File: public/js/icinga.js [hunk 8]
```diff
@@ -235,6 +236,7 @@
                     window.Icinga = oldIcinga;
                     window.icinga = new Icinga(oldConfig);
                     window.icinga.ui.reloadCss();
+                    window.icinga.ui.reloadJs();
```
Comment (lippserd) [inline public/js/icinga.js]: Again, why?

File: public/js/icinga/loader.js [hunk 9]
```diff
@@ -615,6 +615,11 @@
                 this.icinga.ui.reloadCss();
             }
 
+            var jsreload = req.getResponseHeader('X-Icinga-Reload-Js');
+            if (jsreload) {
```
Comment (lippserd) [inline public/js/icinga/loader.js]: You don't need the extra variable.

File: public/js/icinga/ui.js [hunk 10]
```diff
@@ -128,6 +128,12 @@
             });
         },
 
+        reloadJs: function () {
+            var icinga = this.icinga;
```
Comment (lippserd) [inline public/js/icinga/ui.js]: You don't need the extra variable.

File: library/Icinga/Web/Response.php (hunk 6 above)
Comment (nmartinii) [reply library/Icinga/Web/Response.php]: Should I fix this too in any other PHPDoc in this file?

File: library/Icinga/Web/Response.php (hunk 5 above)
Comment (nmartinii) [reply library/Icinga/Web/Response.php]: In Response.php there are other cases like this. Should i add newlines too?

**Comment by nilmerg** [comment]
Superseded by #4350

**Pull request closed without merging**